src/widget/wstrobe.cpp

```diff
@@ -0,0 +1,355 @@
+            context.getPixmapSource(context.selectNode(node, "PathDot33Pos")),
+            context.getScaleFactor());
+    m_pDotUnityImage = WImageStore::getImage(
+            context.getPixmapSource(context.selectNode(node, "PathDotUnity")),
+            context.getScaleFactor());
+    m_pDot33NegImage = WImageStore::getImage(
+            context.getPixmapSource(context.selectNode(node, "PathDot33Neg")),
+            context.getScaleFactor());
+
+    m_pPlay = new ControlProxy(
+            m_group, "play", this);
+    m_pPlayPos = new ControlProxy(
+            m_group, "playposition", this);
+    m_pVisualPlayPos = VisualPlayPosition::getVisualPlayPosition(m_group);
+    m_pTrackSamples = new ControlProxy(
+            m_group, "track_samples", this);
+    m_pTrackSampleRate = new ControlProxy(
+            m_group, "track_samplerate", this);
+
+    m_pScratchToggle = new ControlProxy(
+            m_group, "scratch_position_enable", this);
+    m_pScratchPos = new ControlProxy(
+            m_group, "scratch_position", this);
+}
+
+void WStrobe::slotLoadTrack(TrackPointer pTrack) {
+    if (m_loadedTrack) {
+        disconnect(m_loadedTrack.get(), SIGNAL(coverArtUpdated()), this, SLOT(slotTrackCoverArtUpdated()));
+    }
+    m_loadedTrack = pTrack;
+    if (m_loadedTrack) {
+        connect(m_loadedTrack.get(), SIGNAL(coverArtUpdated()), this, SLOT(slotTrackCoverArtUpdated()));
+    }
+}
+
+void WStrobe::slotLoadingTrack(TrackPointer pNewTrack, TrackPointer pOldTrack) {
+    Q_UNUSED(pNewTrack);
+    if (m_loadedTrack && pOldTrack == m_loadedTrack) {
+        disconnect(m_loadedTrack.get(), SIGNAL(coverArtUpdated()), this, SLOT(slotTrackCoverArtUpdated()));
+    }
+    m_loadedTrack.reset();
+    update();
+}
+
+void WStrobe::paintEvent(QPaintEvent* e) {
+    Q_UNUSED(e);
+}
+
+void WStrobe::render(VSyncThread* vSyncThread) {
+    if (!isValid() || !isVisible()) {
+        return;
+    }
+
+    auto window = windowHandle();
+    if (window == nullptr || !window->isExposed()) {
+        return;
+    }
+
+    if (!m_pVisualPlayPos.isNull() && vSyncThread) {
+        m_pVisualPlayPos->getPlaySlipAtNextVSync(vSyncThread, &m_dAngleCurrentPlaypos, &m_dGhostAngleCurrentPlaypos);
+    }
+
+    if (m_dAngleCurrentPlaypos != m_dAngleLastPlaypos) {
+        m_fAngle = calculateAngle(m_dAngleCurrentPlaypos);
+        m_dAngleLastPlaypos = m_dAngleCurrentPlaypos;
+    }
+
+    qDebug() << m_fAngle << m_dAngleCurrentPlaypos;
+
+    QPainter p(this);
+    p.setRenderHint(QPainter::Antialiasing);
+    p.setRenderHint(QPainter::HighQualityAntialiasing);
+    p.setRenderHint(QPainter::SmoothPixmapTransform);
+
+    double smallHight = height() * 0.23;
+    double bigHight = height() * 0.34;
+    double dot60PosWidth = height() * 0.40 / 216 * 203;
+    double dot33PosWidth = height() * 0.40 / 216 * 209;
+    double dotUnityWidth = height() * 0.40 / 216 * 216;
+    double dot33NegWidth = height() * 0.40 / 216 * 223;
```

Comment:
Can you add some comments that explain these values and calculations? This also applies to the calculations below.